File: binance4j-wallet/src/main/java/com/binance4j/wallet/tradefee/TradeFee.java
package com.binance4j.wallet.tradefee;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Data;

/** A trading pair trade fees */
@Data
@JsonAutoDetect(fieldVisibility = Visibility.ANY)
@JsonIgnoreProperties(ignoreUnknown = true)
public class TradeFee {
    /**
     * The trading pair
     */
    private String symbol;
    /**
     * The maker fees (limit order)
     */
    private BigDecimal makerCommission;
    /**
     * The taker fees (market order)
     */
    private BigDecimal takerCommission;
}

File: connectors/src/main/kotlin/com/binance4j/wallet/dto/TradeFee.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.wallet.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * A trading pair trade fees.
 *
 * @property symbol Trading pair.
 * @property makerCommission Maker fees (limit order).
 * @property takerCommission Taker fees (market order).
 */
@ApiModel("A trading pair trade fees.")
data class TradeFee(
	@ApiModelProperty("Trading pair.")
	val symbol: String,
	@ApiModelProperty("Maker fees (limit order).")
	val makerCommission: String,
	@ApiModelProperty("Taker fees (market order).")
	val takerCommission: String
)
